src/ecredis.erl

```diff
@@ -0,0 +1,420 @@
+get_key_from_command([Term1,Term2|Rest]) when is_bitstring(Term2) ->
+    get_key_from_command([Term1,bitstring_to_list(Term2)|Rest]);
+get_key_from_command([Term1,Term2|Rest]) ->
+    case string:to_lower(Term1) of
+        "info" ->
+            undefined;
+        "config" ->
+            undefined;
+        "shutdown" ->
+            undefined;
+        "slaveof" ->
+            undefined;
+        "eval" ->
+            get_key_from_rest(Rest);
+        "keys" ->
+            keys;
+        "evalsha" ->
+            get_key_from_rest(Rest);
+        _ ->
+            Term2
+    end;
+get_key_from_command(_) ->
+    undefined.
+
+%% =============================================================================
+%% @doc Get key for command where the key is in th 4th position (eval and
+%% evalsha commands)
+%% @end
+%% =============================================================================
+-spec get_key_from_rest([anystring()]) -> string() | undefined.
+get_key_from_rest([_,KeyName|_]) when is_bitstring(KeyName) ->
+    bitstring_to_list(KeyName);
+get_key_from_rest([_,KeyName|_]) when is_list(KeyName) ->
+    KeyName;
+get_key_from_rest(_) ->
+    undefined.
+
+%% =============================================================================
+%% @doc Return the hash slot from the key
+%% @end
+%% =============================================================================
+-spec get_key_slot(Key::any()) -> Slot::integer().
+get_key_slot(K) ->
+    %% cast any type of key to a list.
+    Key = lists:concat([K]),
+    KeyToBeHashed = case string:chr(Key,${) of
+                        0 ->
+                            Key;
+                        Start ->
+                            case string:chr(string:substr(Key,Start+1),$}) of
+                                0 ->
+                                    Key;
+                                Length ->
+                                    if
+                                        Length =:= 1 ->
+                                            Key;
+                                        true ->
+                                            string:substr(Key,Start+1,Length-1)
+                                    end
+                            end
+                    end,
+    eredis_cluster_hash:hash(KeyToBeHashed).
+
+
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+% To manage list of eredis connections
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+create_ets_table(ClusterName) ->
+    ets:new(ets_table_name(ClusterName), [protected, set, named_table, {read_concurrency, true}]).
+
+%% Looks up existing redis client pid using Ip, Port and return Pid of the client
+%% connection. If no such connection is present, this method establishes the connection.
+-spec lookup_eredis_pid(ClusterName::atom(), Ip::string(), Port::integer()) -> {ok, Pid::pid()}.
+lookup_eredis_pid(ClusterName, Ip, Port) ->
```

Comment:
nit: I would have passed both Ip and port as a tuple: since functions that are named with lookup usually expected a single key!